warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/BulkIngestKeyAggregatingReducerTest.java

```diff
@@ -0,0 +1,579 @@
+    private int expectedDuplicateKey;
+    private int expectedR1Counter;
+    private int expectedR2Counter;
+    private int expectedR3Counter;
+    private int expectedTab1Counter;
+    private int expectedTab2Counter;
+    private int expectedTab3Counter;
+    private int expectedCombinerCounter;
+    private int expectedDupCounter;
+    
+    private TaskID taskID;
+    private TaskAttemptID taskAttemptID;
+    
+    private enum ExpectedValueType {
+        NO_VALUE, FIRST_VALUE, COMBINED_VALUES, ALL_VALUES
+    }
+    
+    @Before
+    public void setup() throws Exception {
+        confMap = new HashMap();
+        expected = ArrayListMultimap.create();
+        output = ArrayListMultimap.create();
+        duplicateKey = (Counter) new GenericCounter();
+        duplicateKey = (Counter) new GenericCounter();
+        r1Counter = (Counter) new GenericCounter();
+        r2Counter = (Counter) new GenericCounter();
+        r3Counter = (Counter) new GenericCounter();
+        tab1Counter = (Counter) new GenericCounter();
+        tab2Counter = (Counter) new GenericCounter();
+        tab3Counter = (Counter) new GenericCounter();
+        combinerCounter = (Counter) new GenericCounter();
+        dupCounter = (Counter) new GenericCounter();
+        
+        expectedDuplicateKey = 0;
+        expectedR1Counter = 0;
+        expectedR2Counter = 0;
+        expectedR3Counter = 0;
+        expectedTab1Counter = 0;
+        expectedTab2Counter = 0;
+        expectedTab3Counter = 0;
+        expectedCombinerCounter = 0;
+        expectedDupCounter = 0;
+        
+        conf = (Configuration) PowerMockito.mock(Configuration.class);
+        PowerMockito.whenNew(Configuration.class).withAnyArguments().thenReturn(conf);
+        PowerMockito.when(conf.iterator()).thenReturn(confMap.entrySet().iterator());
+        PowerMockito.when(conf.getBoolean(Mockito.eq(CONTEXT_WRITER_OUTPUT_TABLE_COUNTERS), Mockito.eq(false))).thenReturn(false);
+        PowerMockito.doReturn(BulkContextWriter.class).when(conf).getClass(Mockito.eq(CONTEXT_WRITER_CLASS), Mockito.any(), Mockito.any());
+        
+        PowerMockito.mockStatic(TableConfigurationUtil.class, new Class[0]);
+        PowerMockito.when(TableConfigurationUtil.getTables((Configuration) Mockito.any(Configuration.class))).thenReturn(tables);
+        
+        context = (TaskInputOutputContext<BulkIngestKey,Value,BulkIngestKey,Value>) PowerMockito.mock(TaskInputOutputContext.class);
+        PowerMockito.doAnswer(invocation -> {
+            BulkIngestKey k = invocation.getArgument(0);
+            Value v = invocation.getArgument(1);
+            output.put(k, v);
+            return null;
+        }).when(context).write(Mockito.any(BulkIngestKey.class), Mockito.any(Value.class));
+        PowerMockito.when(context.getCounter(IngestOutput.DUPLICATE_KEY)).thenReturn(duplicateKey);
+        
+        reducer = new BulkIngestKeyAggregatingReducer<>();
+        // reducer.setupContextWriter(conf);
```

Comment:
remove this commented code if it's not needed

Reply:
Removed in latest commit.